### ISSUE TITLE
Update ExpiryTextInput component

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#1155]
## Is your feature request related to a problem? Please describe.

New `DataTable`, new `ExpiryTextInput`.

## Describe the solution you'd like

Couple of options:
- Rewrite `ExpiryTextInput` using a functional stateless component and hooks.
- https://github.com/benhurott/react-native-masked-text use a package!
- Use a date picker https://facebook.github.io/react-native/docs/datepickerandroid

## Describe alternatives you've considered

- Leave it alone

## Additional context

Can you do this without state? Also since we are updating on text changed event, the dispatch would most likely have to reject arbitrary dates? And have `data` be inconsistent with `backingData`. I think the date picker option is actually a better UX and nicer to code 🤷‍♀

**Comment by Chris-Petty** [comment]
I want to get @andreievg 2cents cause it's his little bb

**Comment by joshxg** [comment]
After a little think: a component using a custom hook `useExpiryMask`:

```
const useExpiryMask = (value) => {
    const [expiryValue, setExpiryValue] = useState('')
    
    const wrapSetExpiryValue = (value) => {
        // logic to set a valid date
    }

    return [expiryValue, wrapSetExpiryValue];

}
```

In `ExpiryTextInput`, call `dispatch` when `onEndEditing` callback triggers

**Comment by Chris-Petty** [comment]
> After a little think: a component using a custom hook `useExpiryMask`:
> 
> ```
> const useExpiryMask = (value) => {
>     const [expiryValue, setExpiryValue] = useState('')
>     
>     const wrapSetExpiryValue = (value) => {
>         // logic to set a valid date
>     }
> 
>     return [expiryValue, wrapSetExpiryValue];
> 
> }
> ```
> 
> In `ExpiryTextInput`, call `dispatch` when `onEndEditing` callback triggers
> 
> Advantage of hook: Could use in other inputs

Wouldn't `<DatePickerCell />` be more react-y?

**Comment by joshxg** [comment]
Sure - I guess I meant use the hook inside of a 'date' cell

**Comment by Chris-Petty** [comment]
Fixed in #1161
